### PR TITLE
fix(frontend):  SQL server 数据库规格配置及账号规则授权

### PR DESCRIPTION
**Pull request by styLjc**

### CODE REVIEW

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/blueking-dbm?pullRequest=3273) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bk-cla.bktencent.com:443/TencentBlueKing/blueking-dbm?pullRequest=3273) before we can accept your contribution.<br/><hr/>**v_ljcgliu(刘江城)** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bk-cla.bktencent.com:443/check/TencentBlueKing/blueking-dbm?pullRequest=3273) it.</sub>

File: dbm-ui/frontend/src/services/model/sqlserver/sqlserver-base.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import dayjs from 'dayjs';
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/services/model/sqlserver/sqlserver-base.ts]: 这个放services/util/timeBaseClass.ts，类型也改下

File: dbm-ui/frontend/src/services/source/sqlserverPermission.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+      domain: 'sqlserver.asasd.dba.db',
+      version: '7.10.2',
+      proxy_version: '',
+    },
+  },
+];
+
+// 获取授权规则列表
+export function getAccountRulesList() {
+  return Promise.resolve(defalutList.map(item => new SqlserverPermissionModel(item)));
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/services/source/sqlserverPermission.ts:110]: 请求方法要写完整，包括请求路径、返回类型这些，这些静态数据你调试好了就可以删掉

File: dbm-ui/frontend/src/types/auto-imports.d.ts [hunk 3]
```diff
@@ -67,4 +67,5 @@ declare global {
 declare global {
   // @ts-ignore
   export type { Component, ComponentPublicInstance, ComputedRef, ExtractDefaultPropTypes, ExtractPropTypes, ExtractPublicPropTypes, InjectionKey, PropType, Ref, VNode, WritableComputedRef } from 'vue'
+  import('vue')
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/types/auto-imports.d.ts]: 这个丢弃，不用提交

File: dbm-ui/frontend/src/views/service-apply/create-db-module/SqlServerCreateDbModule.vue [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+  });
+
+  const { data: operatingSystemVersionList } = useRequest(getSqlServerSystemVersion, {
+    defaultParams: [{
+      sqlserver_version: 'MSSQL_Enterprise_2008',
+    }],
+  });
+
+  watch(
+    () => formData.version, (version) => {
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/service-apply/create-db-module/SqlServerCreateDbModule.vue]: 这里不用换行

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/SqlServer.vue [hunk 5]
```diff
@@ -0,0 +1,719 @@
...
+            required>
+            <BkSelect
+              v-model="formData.details.db_module_id"
+              class="item-input"
+              :clearable="false"
+              filterable
+              :input-search="false"
+              style="display: inline-block;">
+              <BkOption
+                v-for="(item) in moduleList"
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/SqlServer.vue]: 去掉括号

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/components/PreviewTable.vue [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+      if (props.isShowNodes) {
+        singleColumns.push({
+          label: t('服务器'),
+          field: 'backend',
+          width: 200,
+          rowspan: () => props.data.length || 1,
+          render: () => {
+            const hosts = props.nodes.backend;
+            return (
+            <div class="host-list">
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/components/PreviewTable.vue]: 加空格

File: dbm-ui/frontend/src/views/sqlserver-manage/apply/components/PreviewTable.vue [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+      {
+        label: t('字符集'),
+        field: 'charset',
+        showOverflowTooltip: true,
+      },
+    ];
+    return haColumns;
+  });
+
+  const setCellClass = ({ field }: { field: string }) => (['backend'].includes(field) ? 'host-td' : '');
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/apply/components/PreviewTable.vue]: 一个就没必要用数组转一下了，直接判断是否相等即可

File: dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue [hunk 8]
```diff
@@ -0,0 +1,672 @@
...
+  const isCN = computed(() => locale.value === 'zh-cn');
+
+  const tableOperationWidth = computed(() => {
+    if (!isStretchLayoutOpen.value) {
+      return isCN.value ? 270 : 420;
+    }
+    return 100;
+  });
+
+  const { run: runCreateTicket } = useRequest(createTicket, { manual: true });
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue]: manual那里换行

File: dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue [hunk 9]
```diff
@@ -0,0 +1,672 @@
...
+    tableData: SqlServerHaClusterModel[],
+  ) => {
+    const copyList: copyListType[] =  tableData.reduce((acc, cluster) => {
+      const masterDetails = cluster.masters.map(extractServerDetails);
+      const slaveDetails = cluster.slaves.map(extractServerDetails);
+      return acc.concat(masterDetails, slaveDetails);
+    }, [] as copyListType[]);
+
+    if (copyList.length) {
+      const instances = copyList.map(item => (isInstance ? `${item.instance}` : `${item.ip}`));
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue]: `${}`是否多余

File: dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue [hunk 10]
```diff
@@ -0,0 +1,672 @@
...
+      copy(instances.join('\n'));
+    } else {
+      messageWarn(isInstance ? t('没有可复制实例') : t('没有可复制IP'));
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue]: 参数最好不要写三目运算的形式，放外面用一个常量装起来好一些

File: dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue [hunk 11]
```diff
@@ -0,0 +1,672 @@
...
+    }
+    return classStack.join(' ');
+  };
+
+  /**
+   * 查看详情
+   */
+  const handleToDetails = (
+    data: SqlServerHaClusterModel,
+    isAllSpread:boolean = false,
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/List.vue]: :boolean 多余

File: dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/detail/BaseInfo.vue [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    :data="data" />
+</template>
+
+<script setup lang="ts">
+  import { useI18n } from 'vue-i18n';
+  import { useRequest } from 'vue-request';
+
+  import { getHaClusterDetail } from '@services/source/sqlserveHaCluster';
+
+  import
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/ha-cluster-list/components/detail/BaseInfo.vue]: import Editinfo, { \n

File: dbm-ui/frontend/src/views/sqlserver-manage/permission/components/CreateRule.vue [hunk 13]
```diff
@@ -0,0 +1,389 @@
...
+      messageSuccess(t('成功添加授权规则'));
+      emits('success');
+      window.changeConfirm = false;
+      handleClose();
+    },
+  });
+
+  const handleSelectAllPrivileges = (checked: boolean) => {
+    checkAllPrivileges.value = checked;
+    checked && (formData.privilege = []);
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/permission/components/CreateRule.vue]: if () {}

File: dbm-ui/frontend/src/views/sqlserver-manage/permission/index.vue [hunk 14]
```diff
@@ -0,0 +1,536 @@
...
+      label: t('账号名称'),
+      field: 'user',
+      showOverflowTooltip: false,
+      render: ({ data }: { data: PermissionRuleExtend }) => (
+        <div
+          class="permission-rules-cell"
+          onClick={ () => handleToggleExpand(data) }>
+            {
+              data.rules.length > 1 && (
+                <i class={ ['db-icon-down-shape user-icon', { 'user-icon-expand': data.isExpand }] } />
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/permission/index.vue]: 换行

File: dbm-ui/frontend/src/views/sqlserver-manage/permission/index.vue [hunk 15]
```diff
@@ -0,0 +1,536 @@
...
+  const authorizeDbs = ref();
+
+  const {
+    data,
+    loading: isLoading,
+    run: runAccountRulesList,
+  } = useRequest(getAccountRulesList, {
+    manual: true,
+    onSuccess() {
+      data.value = data.value && data.value.map(item => (Object.assign(item, { isExpand: item.rules.length > 1 })));
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/permission/index.vue]: 用onSuccess(result) {}，后面要用到返回数据的地方取result

File: dbm-ui/frontend/src/views/sqlserver-manage/permission/index.vue [hunk 16]
```diff
@@ -0,0 +1,536 @@
...
+  });
+
+  runAccountRulesList();
+
+  const handleSearch = () => {
+    nextTick(runAccountRulesList);
+  };
+
+  // 设置行样式
+  const setRowClass = (row: PermissionRuleExtend) => (row.isNew && 'is-new');
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/permission/index.vue]: row.isNew ? 'is-new' : ''

File: dbm-ui/frontend/src/views/sqlserver-manage/permission/index.vue [hunk 17]
```diff
@@ -0,0 +1,536 @@
...
+
+  /**
+   * 列表项展开/收起
+   */
+  const handleToggleExpand = (data: PermissionRuleExtend) => {
+    if (data.rules.length <= 1) {
+      return;
+    }
+    // eslint-disable-next-line no-param-reassign
```
Comment (jinquantianxia) [inline dbm-ui/frontend/src/views/sqlserver-manage/permission/index.vue]: Object.assign

**Pull request closed without merging**